rt/rs/description-swagger-ui/src/main/java/org/apache/cxf/jaxrs/swagger/ui/SwaggerUiConfig.java

```diff
@@ -260,6 +261,10 @@ public SwaggerUiConfig tryItOutEnabled(boolean tryItOut) {
     }
 
     public Boolean isQueryConfigEnabled() {
+        if (!Boolean.TRUE.equals(queryConfigEnabled) && !getConfigParameters().isEmpty()) {
```

Comment:
@awrb this is a getter, which is not supposed to modify any other field, the `queryConfigEnabled` is 3 state property: `null`, `true`, `false`. We should only change this property implicitly when its value is not provided (means, it is `null`). If it is set to `true` or `false` already, we have to respect that.

Reply:
Good point @reta, where do you think is the best place to modify it implicitly? The other place I put the modification was in `getConfigParameters()`, but it's also a kind of getter (just an aggregative one) and you probably wouldn't expect it to modify fields. An alternative would be in each setter (other than  `setQueryConfigEnabled`), but the drawback would be some repetition, e.g:
```
 public void setUrl(final String url) {
        this.url = url;
        setQueryConfigEnabledIfNecessary();
    }

public void setMaxDisplayedTags(Integer maxDisplayedTags) {
        this.maxDisplayedTags = maxDisplayedTags;
        setQueryConfigEnabledIfNecessary();
    }

private void setQueryConfigEnabledIfNecessary() {
   // perhaps also check if a sensible, non-blank value was set?
   if (this.queryConfigEnabled == null) {
      setQueryConfigEnabled(true);
   }
}
```

Reply:
> An alternative would be in each setter

Thanks @awrb  I think repetition is not as bad, it is still isolated  to a function call (and this is one of the reasons setters are there for, to have an ability to introduce some hooks if necessary).